src/Hintman/Hint/NoNewlineAtFileEnd.hs

```diff
@@ -0,0 +1,59 @@
+module Hintman.Hint.NoNewlineAtFileEnd
+       ( getNoNewlineAtFileEndComments
+       ) where
+
+import Hintman.Core.Hint (Hint, HintType (NoNewlineAtFileEnd), Line (..), simpleSuggestion, toLines)
+import Hintman.Core.PrInfo (ModifiedFile (..))
+import Hintman.Core.Review (Comment (..), createComment)
+
+import qualified Data.Text as T
+
+
+-- | Creates 'Comment's on adding newlines in all modified files.
+getNoNewlineAtFileEndComments :: [ModifiedFile] -> [Comment]
+getNoNewlineAtFileEndComments = mapMaybe getFileNoNewlineAtFileEndComment
+
+-- | Takes a file and returns a comment if file doesn't end with a '\n'.
+getFileNoNewlineAtFileEndComment :: ModifiedFile -> Maybe Comment
+getFileNoNewlineAtFileEndComment mf@ModifiedFile{..} = spawnComment $
+    maybeGetNonEmptyLastLine fileLines diffText
```

Comment:
I'm starting to worry that this becomes a bit inefficient and slow, especially for big files. Now it's an additional traversal through all lines of each file. It will be much nicer to combine with existing traversals.

Reply:
Yeah, I agree. But at the same time, I like the approach when all the lints are sort-of independent and unaware of other lints.

What do you think about extending `ModifiedFile` with another field, which will contain already splitted lines, named e.g. `mfLines`?

This, however, will add an additional invariant assuming that file content should be valid `utf-8`, but since we already using `decodeUtf8` without preemptive checks, I think that's not a big problem. More than that, then we'll be able to add such a check (if needed) in just one place.

Reply:
Other question is do you want me to implement this as a part of the current PR, or should I open a new PR for that?

Reply:
I guess later we can split the content in the main function and then just give to the check functions only those parts that they will need for the check. But this is definitely the separate task to be done.

Reply:
I think that splitting file content only once can be fast. In addition instead of `[Line]` we can use `Vector Line` to have constant access to elements by index. So we can create this structure only once and then pass it everywhere. But as @vrom911 noticed, this is definitely will be in a separate PR.

Reply:
Created #157 for this.